Documentation about makefile (#545)
mattermost/mattermost-developer-documentation: site/content/contribute/desktop/developer-setup.md

- NPM
- Git

### Windows

> Recommended way to install prerequisites on windows is to install chocolatey package manager.

Reviewer: ```suggestion
The recommended way to install the required components on Windows is to install the Chocolatey package manager.
```

> once installed run from an administrator powershell console

Reviewer: ```suggestion
Once it's installed, run the following commands in the PowerShell console as `administrator`. 
```

```
choco install nvm
```
```
choco install git
```

> from a non privileged powershell console:

Reviewer: ```suggestion
If you're using PowerShell without privileged access, run:
```

```
nvm install 10.18.1
```

> ## Installing dependencies

Reviewer: ```suggestion
## Installing Dependencies
```

> After installation, dependencies of `src/` directory are also installed.

Reviewer: ```suggestion
When installation is complete, dependencies of the`src/` directory are also installed.
```

### OSX & Linux
```
$ npm install
```
### Windows

> on an administrator powershell run

Reviewer: ```suggestion
Run the following command in the PowerShell console as `administrator`. 
```

```
> .\scripts\Makefile.ps1 Install-Deps
```

> #### Known problems

Reviewer: ```suggestion
#### Known Limitations
```

Author: isn't this a usual section for manuals? anyway, this is not a limitation but issues I encountered while running the above commands. With this section we aim to provide workarounds for people in case they run into the same problem.

Reviewer: That makes a lot of sense, so I've changed it to "Known Problems". :)

##### Policy

> If the above command fails, it might be due to not having the right policy to install. In that case use this command to temporarily change the policy:

Reviewer: ```suggestion
If the above command fails, it might be due to not having the right policy to install. You can temporarily change the policy using:
```

```
> Set-ExecutionPolicy Bypass -Scope Process -Force;
```

##### pngquant-bin

> On windows there is no default libpng-dev available, so far the best option is to install visual studio community, which will provide dependencies to build the required package.

Reviewer: ```suggestion
On Windows there is no default `libpng-dev` available. Currently the best option is to install Visual Studio Community, which provides the dependencies required to build the package.
```

Files are defined in `electron-builder.json`.
Packages will be generated into `release/` directory.

> ### OSX & Linux

Reviewer: ```suggestion
### macOS and Linux
```
